enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -99,259 +116,217 @@ This is often required for industries such as the US public sector, healthcare o
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
 
-#### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
+* Server-authenticated TLS is enabled if `destination` is a secure URL (e.g. 'https:')
+* Client-authenticated TLS is enabled if `destination` is secure *and* `secretRef` has keys `tls.crt`, `tls.key`, `ca-bundle.crt`
+  - it is an error if `secretRef` is present but `destination` is *not* secure, or the required 'tls.' keys are missing or invalid.
+* An intentionally insecure output (no TLS) must have `insecure: true`
+  - avoid accidental insecure configuration of an output that was intended to be secure.
+* The user is responsible for creating and maintaining the secret objects
+* The cluster logging operator is responsible for watching secrets and applying changes,  e.g. if the user replaces a certificate or changes a password in a secret, the operator must re-connect affected outputs with the new credentials.
+
+#### Reliability
+* Output connections will automatically re-connect on disconnect.
+* Changes to secrets will trigger automatic re-connect with new credentials.
+
+#### Scale
+* The cluster logging operator generates configuration for the collector
+  - with a singleton ClusterLogForwarder this is unlikely to be a scaling problem.
+* The actual connection and forwarding to remote endpoints is done by the collector
+  - we rely on the collector to scale and perform.
 
-For example, If you have specified a pipeline for `logs.app` sources, but did not specify one for `logs.infra`, then `logs.infra` will be dropped
+#### Metrics
 
+Note: the actual forwarding is done by the collector, so we can only provide metrics that are available from the collector.
+
+Relevant metrics include:
+
+* Counter:
+  - Volume (bytes) per input, per pipeline, per output and total.
+* Histogram/Summary:
+  - Throughput (bytes/sec) per input, per pipeline, per output and total.
+  - Read size: per input
+  - Write size: per output
+  - Latency (sec)
+    - per pipeline: from read to written on all outputs.
+    - per output: from read to written on this output.
+
+#### Cluster Logging Operator
+
+The `cluster-logging-operator` will use the `ClusterLogForwarder` configuration to:
+
+* generate output configuration for the collector that respects all the pipelines.
+* mount secrets in the collector daemonset as needed for each endpoint.
+  - `/var/run/ocp-collector/secrets/<secretName>/{tls.crt,tls.key,ca-bundle.crt,username,password}`
```

Comment:
Do we need to account for namespace  if we intend to support it in future?  Honestly I'm not sure how we would do that in future anyway since it seems like a broad permission to give to the CLO.  Would we 'monitor' and copy to 'openshift-logging'?

Reply:
I think it would make it easier for us to restrict this to a single namespace. Otherwise its as you're eluding to -- we need to maintain permissions to any namespace that this secret could exist in, I'm not sure it would be advantageous to copy it to `openshift-logging` if we can already read it elsewhere, it feels like it'd just introduce another point of failure (sync-ing secrets)

Reply:
For GA this will be restricted to openshift namespace as it is  now.  For the future, this path is  an implementation detail (only read by collectors we configure) so we can restructure if needed to cope with whatever flexibility we decide to provide.